src/NativeScript/ManualInstrumentation.h

```diff
@@ -0,0 +1,77 @@
+//
+// Created by Panayot Cankov on 26/05/2017.
+//
+
+#ifndef MANUALINSTRUMENTATION_H
+#define MANUALINSTRUMENTATION_H
+
+#import <chrono>
+#include <string>
+#include <WTF/Assertions.h>
+
+namespace tns {
+    namespace instrumentation {
+        enum Mode {
+            Disabled = 0,
+            Uninitialized = 1,
+            Enabled = 2
+        };
+        class Frame {
+        public:
+            inline Frame() : Frame("") { }
+            inline Frame(std::string name) : start(mode == Mode::Disabled ? disabled_time : std::chrono::steady_clock::now()), name(name) {}
+            inline Frame(const Frame &copy) : start(copy.start), name(copy.name) {}
+            
+            inline ~Frame() {
+                if (!name.empty() && check()) {
+                    log(name);
+                }
+            }
+            
+            inline bool check() const {
+                if (mode == Mode::Disabled) {
+                    return false;
+                }
+                std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+                auto duration = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::operator-(end, start)).count();
+                return duration >= 16000;
```

Comment:
It means we are not interested in fast frames and don't want to large amount of small frames. Logging may be slow.